learn/urls.py

```diff
@@ -3,60 +3,65 @@
 
-	url(r'^(?P<topic_slug>[-\w]+)/(?P<resource_slug>[-\w]+)/(?P<action>[-\w]+)/',managevote,name="vote"),
+	path('<slug:topic_slug>/<slug:resource_slug>/add-review/',ReviewCreate.as_view(),name="ReviewCreate"),
+	path('<slug:topic_slug>/<slug:resource_slug>/update-review/<int:pk>/',ReviewUpdate.as_view(),name="ReviewUpdate"),
+	
+	path('<slug:topic_slug>/<slug:resource_slug>/delete-review/<int:pk>/',ReviewDelete.as_view(),name="ReviewDelete"),
 
-	url(r'^category/all/$',CategoryList.as_view(),name="CategoryList"),
-	url(r'^category/(?P<category_slug>[-\w]+)/$',SelectedTopicList.as_view(),name="SelectedTopicList"),
+	path('<slug:topic_slug>/<slug:resource_slug>/<slug:action>/',managevote,name="vote"),
 
-	url(r'^ajaxcall/topicautocomplete/',autocompleteSuggestionTopic,name="AjaxTopicAutocomplete"),
+	path('category/all/',CategoryList.as_view(),name="CategoryList"),
+	path('category/<slug:category_slug>/',SelectedTopicList.as_view(),name="SelectedTopicList"),
+	path('ajaxcall/topicautocomplete/',autocompleteSuggestionTopic,name="AjaxTopicAutocomplete"),
 
 ]
 
 urlpatterns+=[
-	url(r'^login/',LoginView.as_view(redirect_authenticated_user=True),name="login"),
-	url(r'^logout/',LogoutView.as_view(),name="logout"),		
-	url(r'signup/',SignupView.as_view(),name="signup"),
+	path('login/',LoginView.as_view(redirect_authenticated_user=True),name="login"),
+	path('logout/',LogoutView.as_view(),name="logout"),
+	path('signup/',SignupView.as_view(),name="signup"),
 
-	url(r'^myaccount/change_password',PasswordChangeView.as_view(),name="change_password"),
-	url(r'^myaccount/password_change_done/',PasswordChangeDoneView.as_view(),name="password_change_done"),
+	path('myaccount/change_password',PasswordChangeView.as_view(),name="change_password"),
+	
+	path('myaccount/password_change_done/',PasswordChangeDoneView.as_view(),name="password_change_done"),
 
-	url(r'^forgot_password/$',strictly_no_login(PasswordResetView.as_view()),name="forgot_password"),
-	url(r'^password_reset/done/$', password_reset_done, name='password_reset_done'),
+	path('forgot_password/',strictly_no_login(PasswordResetView.as_view()),name="forgot_password"),
+	path('password_reset/done/', PasswordResetDoneView.as_view(), name='password_reset_done'),
 
-    url(r'^reset/(?P<uidb64>[0-9A-Za-z_\-]+)/(?P<token>[0-9A-Za-z]{1,13}-[0-9A-Za-z]{1,20})/$',
-        password_reset_confirm, name='password_reset_confirm'),
-    url(r'^reset/done/$', password_reset_complete, name='password_reset_complete'),
+    path('reset/<slug:uidb64>/<str:token>/',PasswordResetConfirmView.as_view, name='password_reset_confirm'),
+    path('reset/done/', PasswordResetCompleteView.as_view(), name='password_reset_complete'),
 ]
 #to avoid errors
 urlpatterns+=[
-	url(r'^(?P<slug>[-\w]+)/$',TopicDetails.as_view(),name="TopicDetails"),
-	url(r'^(?P<slug>[-\w]+)/test.$',test,name="test"),
+	path('<slug:slug>/',TopicDetails.as_view(),name="TopicDetails"),
+	path('<slug:slug>/test.$',test,name="test"),
```

Comment:
There is an warning on this line 
`?: (2_0.W001) Your URL pattern '<slug:slug>/test.$' [name='test'] has a route that contains '(?P<', begins with a '^', or ends with a '$'. This was likely an oversight when migrating to django.urls.path().`
Remove `$` from this statement.